docs/rules/require-named-capture-group.md

```diff
@@ -0,0 +1,43 @@
+# Suggest using named capture group in regular expression (require-named-capture-group)
```

Comment:
About the rule name, we are using `prefer-` word for the rule which suggests an alternative way in practice. (E.g. `prefer-const`, `prefer-spread`, ...)

Therefore, I think `prefer-named-capture-group` is good.

Reply:
I'd like to change, but do we need any discussion from other team member?

Reply:
@g-plane Up to you if you want to wait for more feedback or not- there's no real formal process to "approve a name change". I'm :+1: to the name change myself although I'm okay with the original name.